libraries/AP_RCTelemetry/AP_RCTelemetry.cpp

```diff
@@ -208,7 +209,8 @@ void AP_RCTelemetry::check_sensor_status_flags(void)
         } else if ((_sensor_status_flags & MAV_SYS_STATUS_SENSOR_3D_MAG) > 0) {
             queue_message(MAV_SEVERITY_CRITICAL, "Bad Compass Health");
             check_sensor_status_timer = now;
-        } else if ((_sensor_status_flags & MAV_SYS_STATUS_SENSOR_ABSOLUTE_PRESSURE) > 0) {
+        } else if ((_sensor_status_flags & MAV_SYS_STATUS_SENSOR_ABSOLUTE_PRESSURE) > 0
+                    && AP::dal().baro().num_instances() > 0) {
```

Comment:
Why are we using the DAL here rather than going straight to the baro library?

Reply:
because these warnings come from the EKF which is using the DAL?

Reply:
> because these warnings come from the EKF which is using the DAL?

I can't see how the EKF is involved at all here.  `MAV_SYS_STATUS_SENSOR_ABSOLUTE_PRESSURE` comes from the AP::baro() singleton, nothing to do with the EKF.  We get the flags from the gcs library - nothing to do with the EKF.  We're in the RC telemetry library - nothing to do with the EKF...

As for the sensor status flags present in this method - the bit being checked here should always be false if you have no baros on your system - that's this line:

```
    return ~health & enabled & present;
```

Perhaps the fix is this?

```
diff --git a/libraries/GCS_MAVLink/GCS.cpp b/libraries/GCS_MAVLink/GCS.cpp
index 20601f8e79f..0393f04930b 100644
--- a/libraries/GCS_MAVLink/GCS.cpp
+++ b/libraries/GCS_MAVLink/GCS.cpp
@@ -184,10 +184,12 @@ void GCS::update_sensor_status_flags()
 
 #if AP_BARO_ENABLED
     const AP_Baro &barometer = AP::baro();
-    control_sensors_present |= MAV_SYS_STATUS_SENSOR_ABSOLUTE_PRESSURE;
-    control_sensors_enabled |= MAV_SYS_STATUS_SENSOR_ABSOLUTE_PRESSURE;
-    if (barometer.all_healthy()) {
-        control_sensors_health |= MAV_SYS_STATUS_SENSOR_ABSOLUTE_PRESSURE;
+    if (barometer.num_instances() > 0) {
+        control_sensors_present |= MAV_SYS_STATUS_SENSOR_ABSOLUTE_PRESSURE;
+        control_sensors_enabled |= MAV_SYS_STATUS_SENSOR_ABSOLUTE_PRESSURE;
+        if (barometer.all_healthy()) {
+            control_sensors_health |= MAV_SYS_STATUS_SENSOR_ABSOLUTE_PRESSURE;
+        }
     }
 #endif
```